arbos/l1pricing/l1pricing.go

```diff
@@ -645,27 +647,57 @@ func (ps *L1PricingState) GetPosterInfo(tx *types.Transaction, poster common.Add
 	return am.BigMulByUint(pricePerUnit, units), units
 }
 
-const TxFixedCostEstimate = 140 // assumed maximum size in bytes of a typical RLP-encoded tx, not including its calldata
+// We don't have the full tx in gas estimation, so we assume it might be a bit bigger in practice.
+const estimationPaddingUnits = 16 * params.TxDataNonZeroGasEIP2028
+const estimationPaddingBasisPoints = 100
+
+var randomNonce = binary.BigEndian.Uint64(crypto.Keccak256([]byte("Nonce"))[:8])
+var randomGasTipCap = new(big.Int).SetBytes(crypto.Keccak256([]byte("GasTipCap"))[:4])
+var randomGasFeeCap = new(big.Int).SetBytes(crypto.Keccak256([]byte("GasFeeCap"))[:4])
+var randV = arbmath.BigMulByUint(params.ArbitrumOneChainConfig().ChainID, 3)
+var randR = crypto.Keccak256Hash([]byte("R")).Big()
+var randS = crypto.Keccak256Hash([]byte("S")).Big()
+
+func makeFakeTxForMessage(message core.Message) *types.Transaction {
```

Comment:
it may be worth noting that this fake tx is invalid (signature doesn't check out, intentionally large field values, even the wrong tx type sometimes, etc)